modules/src/ics02_client/msgs/upgrade_client.rs

```diff
@@ -0,0 +1,52 @@
+//! Definition of domain type msg `MsgUpgradeAnyClient`.
+
+use ibc_proto::ibc::core::client::v1::MsgUpgradeClient as RawMsgUpgradeClient;
+use ibc_proto::ibc::core::commitment::v1::MerkleProof;
+
+use crate::ics02_client::client_def::{AnyClientState, AnyConsensusState};
+use crate::ics23_commitment::commitment::CommitmentProofBytes;
+use crate::ics24_host::identifier::ClientId;
+use crate::signer::Signer;
+use crate::tx_msg::Msg;
+
+pub(crate) const TYPE_URL: &str = "/ibc.core.client.v1.MsgUpgradeClient";
+
+/// A type of message that triggers the upgrade of an on-chain (IBC) client.
+#[derive(Clone, Debug, PartialEq)]
+pub struct MsgUpgradeAnyClient {
+    pub client_id: ClientId,
+    pub client_state: AnyClientState,
+    pub consensus_state: AnyConsensusState,
+    pub proof_upgrade_client: MerkleProof,
+    pub proof_upgrade_consensus_state: MerkleProof,
+    pub signer: Signer,
+}
+
+impl Msg for MsgUpgradeAnyClient {
+    type ValidationError = crate::ics24_host::error::ValidationError;
+    type Raw = RawMsgUpgradeClient;
+
+    fn route(&self) -> String {
+        crate::keys::ROUTER_KEY.to_string()
+    }
+
+    fn type_url(&self) -> String {
+        TYPE_URL.to_string()
+    }
+}
+
```

Comment:
We are missing a `impl Protobuf<RawMsgUpgradeClient> for MsgUpgradeAnyClient` and `TryFrom<RawMsgUpgradeClient> for MsgUpgradeAnyClient ` here.

Reply:
I did not add it on purpose as it was not necessary. I find it difficult to understand which messages need which kind of de/serialization support, so I decided to stop adding these methods unless they're clearly needed.

But if you think it's better to just have the methods already (e.g., for consistency sake), I will add them.

Reply:
I'd rather we add them everywhere, both for consistency, but also because `ibc` is library aimed at being used outside of our own `ibc-relayer` use case, and other users might have different needs. Of course, we could still wait until someone raises the issue to us, but in the case of messages, I think it wouldn't hurt adding those instances by default.

Reply:
Good point! 77691624